packages/@expo/fingerprint/e2e/__tests__/updates-test.ts

```diff
@@ -0,0 +1,73 @@
+import spawnAsync from '@expo/spawn-async';
+import fs from 'fs/promises';
+import path from 'path';
+import rimraf from 'rimraf';
+
+import { createProjectHashAsync } from '../../src/Fingerprint';
+
+jest.mock('../../src/sourcer/ExpoConfigLoader', () => ({
+  // Mock the getExpoConfigLoaderPath to use the built version rather than the typescript version from src
+  getExpoConfigLoaderPath: jest.fn(() =>
+    jest
+      .requireActual('path')
+      .resolve(__dirname, '..', '..', 'build', 'sourcer', 'ExpoConfigLoader.js')
+  ),
+}));
+
+describe('updates support', () => {
+  jest.setTimeout(600000);
+  const tmpDir = require('temp-dir');
+  const projectName = 'fingerprint-e2e-updates';
+  const projectRoot = path.join(tmpDir, projectName);
+
+  beforeAll(async () => {
+    rimraf.sync(projectRoot);
+    // Pin the SDK version to prevent the latest version breaking snapshots
+    await spawnAsync('bunx', ['create-expo-app', '-t', 'blank@sdk-51', projectName], {
+      stdio: 'inherit',
+      cwd: tmpDir,
+      env: {
+        ...process.env,
+        // Do not inherit the package manager from this repository
+        npm_config_user_agent: undefined,
+      },
+    });
+
+    // Add appId
+    const appConfigPath = path.join(projectRoot, 'app.json');
+    const appConfig = JSON.parse(await fs.readFile(appConfigPath, 'utf8'));
+    appConfig.expo.android.package = 'dev.expo.fingerprint';
+    appConfig.expo.ios.bundleIdentifier = 'dev.expo.fingerprint';
+    await fs.writeFile(appConfigPath, JSON.stringify(appConfig, null, 2));
+
+    // Update package.json scripts
+    // Temporary workaround because prebuild will change scripts
```

Comment:
this should no longer be needed as of your other PR, correct?

Reply:
correct. #29679 is stacked on top this pr. i'll wait to merge this pr first. otherwise, it's hard to review this pr.